src/Simplify_EQ.cpp

```diff
@@ -58,21 +58,36 @@ Expr Simplify::visit(const EQ *op, ConstBounds *bounds) {
 
     auto rewrite = IRMatcher::rewriter(IRMatcher::eq(delta, 0), op->type, delta.type());
 
-    if (rewrite(c0 == 0, fold(c0 == 0)) ||
-        rewrite(x + c0 == 0, x == fold(-c0)) ||
-        rewrite(c0 - x == 0, x == c0)) {
-        return rewrite.result;
-    }
-
     if (rewrite(broadcast(x) == 0, broadcast(x == 0, lanes)) ||
         (no_overflow(delta.type()) && rewrite(x * y == 0, (x == 0) || (y == 0))) ||
         rewrite(select(x, 0, y) == 0, x || (y == 0)) ||
         rewrite(select(x, c0, y) == 0, !x && (y == 0), c0 != 0) ||
         rewrite(select(x, y, 0) == 0, !x || (y == 0)) ||
-        rewrite(select(x, y, c0) == 0, x && (y == 0), c0 != 0)) {
+        rewrite(select(x, y, c0) == 0, x && (y == 0), c0 != 0) ||
+        rewrite(max(x, y) - y == 0, x <= y) ||
```

Comment:
@jn80842 Please take a look at these new rules and see if I've done something wrong